src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.Fixer.cs

```diff
@@ -0,0 +1,214 @@
+            // offset should become start
+            // count should become length
+            SyntaxNode namedStartNode = isOffsetNamed ? generator.Argument("start", RefKind.None, offsetValueNode) : offsetValueNode;
+            SyntaxNode namedLengthNode = isCountNamed ? generator.Argument("length", RefKind.None, countValueNode) : countValueNode;
+
+            // Generate an invocation of the AsMemory() method from the byte array object, using the correct named arguments
+            SyntaxNode asMemoryExpressionNode = generator.MemberAccessExpression(bufferValueNode, "AsMemory");
+            SyntaxNode asMemoryInvocationNode = generator.InvocationExpression(asMemoryExpressionNode, namedStartNode, namedLengthNode);
+
+            // Generate the new buffer argument, ensuring we include the argument name if the user originally indicated one
+            SyntaxNode namedBufferNode = isBufferNamed ? generator.Argument("buffer", RefKind.None, asMemoryInvocationNode) : asMemoryInvocationNode;
+
+            // Create an async method call for the stream object with no arguments
+            SyntaxNode asyncMethodNode = generator.MemberAccessExpression(streamInstanceNode, methodName);
+
+            // Add the arguments to the async method call, with or without CancellationToken
+            SyntaxNode[] nodeArguments;
+            if (cancellationTokenValueNode != null)
+            {
+                SyntaxNode namedCancellationTokenNode = isCancellationTokenNamed ? generator.Argument("cancellationToken", RefKind.None, cancellationTokenValueNode) : cancellationTokenValueNode;
+                nodeArguments = new SyntaxNode[] { namedBufferNode, namedCancellationTokenNode };
+            }
+            else
+            {
+                nodeArguments = new SyntaxNode[] { namedBufferNode };
+            }
+            SyntaxNode newInvocationExpression = generator.InvocationExpression(asyncMethodNode, nodeArguments);
+
+            SyntaxNode newRoot = generator.ReplaceNode(root, invocation.Syntax, newInvocationExpression);
+            return Task.FromResult(doc.WithSyntaxRoot(newRoot));
+        }
+    }
+
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class PreferStreamAsyncMemoryOverloadsCSharpFixer : PreferStreamAsyncMemoryOverloadsFixer
+    {
+        protected override IArgumentOperation? GetArgumentByPositionOrName(ImmutableArray<IArgumentOperation> args, int index, string name, out bool isNamed)
+        {
+            isNamed = false;
+
+            // The expected position is beyond the total arguments, so we don't expect to find the argument in the array
+            if (index >= args.Length)
+            {
+                return null;
+            }
+            // If the argument in the specified index does not have a name, then it is in its expected position
+            else if (args[index].Syntax is CodeAnalysis.CSharp.Syntax.ArgumentSyntax argNode && argNode.NameColon == null)
```

Comment:
Should be able simplify once this type is moved to C# layer and you can add `using Microsoft.CodeAnalysis.CSharp.Syntax;` at the top of the file.

```suggestion
            else if (args[index].Syntax is ArgumentSyntax argNode && argNode.NameColon == null)
```